spec/pages_spec.rb

```diff
@@ -13,13 +64,28 @@
       visit p
     end
 
-    it 'has only valid internal hyperlinks' do
-      page.all(:css, 'a').each do |link|
-        next if link.text == '' || link[:href].match(/(http|\/\/).*/)
-        page.find(:xpath, link.path).click
-        expect(page.status_code).to be(200), "expected link '#{link.text}' to work"
-        visit p
+    it 'has valid internal hyperlinks' do
+      links = Links.new(page.all(:css, 'a'),urls)
+      links.each do |link|
+        url=link[:href]
+        if !is_uri? url
+          page.find(:xpath, link.path).click
+          expect(page.status_code).to be(200), "expected link '#{link.text}' to work"
```

Comment:
In `be(200)` the usual practice is to replace the magic constant with a `literal` constant (IOW use capitalized variable defined in `spec_helper`.